util/chunk/chunk.go

```diff
@@ -609,6 +609,30 @@ func (c *Chunk) Column(colIdx int) *Column {
 	return c.columns[colIdx]
 }
 
+// GetOrNewColumn returns the colIdx Column in this Chunk or creates a new Column if it's nil.
+func (c *Chunk) GetOrNewColumn(colIdx int, ft *types.FieldType, cap int) *Column {
+	for len(c.columns) < colIdx {
+		c.columns = append(c.columns, nil)
+	}
+	if c.columns[colIdx] == nil {
```

Comment:
I think it will get a panic when `len(c.columns) == colIdx`(line 614-616 will make it).
Your test don't cover the case. 
Or it will never happen in normal situation?

Reply:
You are right, it's a bug.
It is fixed now.